core/drivers/crypto/crypto_api/acipher/rsa.c

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2020 NXP
+ *
+ * Brief   Crypto RSA interface implementation to enable HW driver.
+ */
+#include <drvcrypt.h>
+#include <crypto/crypto.h>
+#include <tee_api_defines_extensions.h>
+#include <tee/tee_cryp_utl.h>
+#include <utee_defines.h>
+
+#include "local.h"
+
+TEE_Result crypto_acipher_alloc_rsa_keypair(struct rsa_keypair *key,
+					    size_t size_bits)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+
+	struct drvcrypt_rsa *rsa = NULL;
+
+	if (!key || !size_bits) {
+		CRYPTO_TRACE("Parameters error (key @%p) (size %zu bits)", key,
+			     size_bits);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	rsa = drvcrypt_get_ops(CRYPTO_RSA);
+	if (rsa)
+		ret = rsa->alloc_keypair(key, size_bits);
+
+	CRYPTO_TRACE("RSA Keypair (%zu bits) alloc ret = 0x%" PRIx32, size_bits,
+		     ret);
+	return ret;
+}
+
+TEE_Result crypto_acipher_alloc_rsa_public_key(struct rsa_public_key *key,
+					       size_t size_bits)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+	struct drvcrypt_rsa *rsa = NULL;
+
+	if (!key || !size_bits) {
+		CRYPTO_TRACE("Parameters error (key @%p) (size %zu bits)", key,
+			     size_bits);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	rsa = drvcrypt_get_ops(CRYPTO_RSA);
+	if (rsa)
+		ret = rsa->alloc_publickey(key, size_bits);
+
+	CRYPTO_TRACE("RSA Public Key (%zu bits) alloc ret = 0x%" PRIx32,
+		     size_bits, ret);
+	return ret;
+}
+
+void crypto_acipher_free_rsa_public_key(struct rsa_public_key *key)
+{
+	struct drvcrypt_rsa *rsa = NULL;
+
+	if (key) {
+		rsa = drvcrypt_get_ops(CRYPTO_RSA);
+		if (rsa) {
+			CRYPTO_TRACE("RSA Public Key free");
+			rsa->free_publickey(key);
+		}
+	}
+}
+
+TEE_Result crypto_acipher_gen_rsa_key(struct rsa_keypair *key, size_t size_bits)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+	struct drvcrypt_rsa *rsa = NULL;
+
+	if (!key || !size_bits) {
+		CRYPTO_TRACE("Parameters error (key @%p) (size %zu bits) ",
+			     key, size_bits);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	rsa = drvcrypt_get_ops(CRYPTO_RSA);
+	if (rsa)
+		ret = rsa->gen_keypair(key, size_bits);
+
+	CRYPTO_TRACE("RSA Keypair (%zu bits) generate ret = 0x%" PRIx32,
+		     size_bits, ret);
+
+	return ret;
+}
+
+TEE_Result crypto_acipher_rsanopad_decrypt(struct rsa_keypair *key,
+					   const uint8_t *cipher,
+					   size_t cipher_len, uint8_t *msg,
+					   size_t *msg_len)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+	struct drvcrypt_rsa *rsa = NULL;
+	struct drvcrypt_rsa_ed rsa_data = {};
```

Comment:
We normally type this initializer as `{ }`, note the space in between. I'm not sure if other care, but for me it stands out a bit (in a negative way) when the space is skipped.

Reply:
I will add a space